src/devices/Common/Iot/Device/Multiplexing/VirtualOutputSegment.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Multiplexing.Utility
+{
+    /// <summary>
+    /// Interface that abstracts multiplexing over a segment of outputs.
+    /// </summary>
+    public sealed class VirtualOutputSegment : IOutputSegment
```

Comment:
To be honest, I'm not sure I understand if there is enough value of having a separate class for this as opposed to just have a PinValue array on the GpioOutputSegment class that preserves state. The only thing I see is so that the WriteByteAsValue methods can get reused by CharlieplexSegment and GpioOutputSegment but do let me know if I'm missing something else. Also, if we do decide to keep it around, can we make it internal? I'm not sure I see the point of having this type public as it is really just an implementation detail.

Reply:
You are right. It's borderline. How about we go with this for the PR, and then decide what to do with it. Perhaps this was a good tool for prototyping quickly, since I didn't have to make changes in multiple places. Certainly, it could be collapsed, but it's an implementation detail so can be updated later.